plugins/extractors/bigquery/bigquery_test.go

```diff
@@ -56,6 +57,12 @@ func TestInit(t *testing.T) {
 	})
 }
 
+func TestExtract(t *testing.T) {
```

Comment:
sorry, forgot to remove this, yes this is a TODO

Reply:
Instead of simply removing it, can we add tests for BQ using https://github.com/goccy/bigquery-emulator? BigQuery is an important extractor that is currently not covered by tests.

Reply:
thanks for the pointer, found the same yesterday,
facing memory leak issue using go imports, will try dockertest approach, hopefully it will works

Reply:
Could you elaborate on the memory leak issue?

Reply:
not sure if it's only happen in my local machine, but you can try it in your local running their example code

https://github.com/goccy/bigquery-emulator#synopsis

when i run the code it try to compile the deps and it goes indefinetely and the memory usage are growing until my machine slowing down
![image](https://github.com/goto/meteor/assets/858000/d56267ba-ab3b-40c2-ac14-9c8b8908cf7b)

```
95015  91.3  8.8 412419696 2968752 s032  R+   12:00PM   0:25.25 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -cc1 -triple arm64-apple-macosx12.0.0 -Wundef-prefix=TARGET_OS_ -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name bind.cc -mrelocation-model pic -pic-level 2 -mframe-pointer=non-leaf -fno-strict-return -fno-rounding-math -funwind-tables=2 -fobjc-msgsend-selector-stubs -target-sdk-version=13.1 -fvisibility-inlines-hidden-static-local-var -target-cpu apple-m1 -target-feature +v8.5a -target-feature +fp-armv8 -target-feature +neon -target-feature +crc -target-feature +crypto -target-feature +dotprod -target-feature +fp16fml -target-feature +ras -target-feature +lse -target-feature +rdm -target-feature +rcpc -target-feature +zcm -target-feature +zcz -target-feature +fullfp16 -target-feature +sm4 -target-feature +sha3 -target-feature +sha2 -target-feature +aes -target-abi darwinpcs -fallow-half-arguments-and-returns -debug-info-kind=standalone -dwarf-version=4 -debugger-tuning=lldb -target-linker-version 820.1 -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/go-zetasql -I /var/folders/4s/kj0rwqjj67sfjnbv5387s2fw0000gp/T/go-build2327985354/b499/ -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/protobuf -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/gtest -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/icu -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/re2 -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/json -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/googleapis -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/flex/src -D HAVE_PTHREAD -D U_COMMON_IMPLEMENTATION -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/protobuf -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/gtest -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/icu -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/re2 -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/json -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/googleapis -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/flex/src -D HAVE_PTHREAD -D U_COMMON_IMPLEMENTATION -I/usr/local/include -stdlib=libc++ -internal-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1 -internal-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -O2 -Wno-char-subscripts -Wno-sign-compare -Wno-switch -Wno-unused-function -Wno-deprecated-declarations -Wno-inconsistent-missing-override -Wno-unknown-attributes -Wno-macro-redefined -Wno-shift-count-overflow -Wno-enum-compare-switch -Wno-return-type -Wno-subobject-linkage -Wno-unknown-warning-option -Wno-char-subscripts -Wno-sign-compare -Wno-switch -Wno-unused-function -Wno-deprecated-declarations -Wno-inconsistent-missing-override -Wno-unknown-attributes -Wno-macro-redefined -Wno-shift-count-overflow -Wno-enum-compare-switch -Wno-return-type -Wno-subobject-linkage -Wno-unknown-warning-option -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -Wno-reserved-identifier -Wno-gnu-folding-constant -Wno-cast-function-type -Wno-bitwise-instead-of-logical -std=c++1z -fdeprecated-macro -fdebug-compilation-dir=/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/go-zetasql -fdebug-prefix-map=/var/folders/4s/kj0rwqjj67sfjnbv5387s2fw0000gp/T/go-build2327985354/b499=/tmp/go-build -ferror-limit 19 -pthread -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fno-cxx-modules -fcxx-exceptions -fexceptions -fmax-type-align=16 -fno-caret-diagnostics -vectorize-loops -vectorize-slp -clang-vendor-feature=+messageToSelfInClassMethodIdReturnType -clang-vendor-feature=+disableInferNewAvailabilityFromInit -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -clang-vendor-feature=+enableAggressiveVLAFolding -clang-vendor-feature=+revert09abecef7bbf -clang-vendor-feature=+thisNoAlignAttr -clang-vendor-feature=+thisNoNullAttr -mllvm -disable-aligned-alloc-awareness=1 -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /var/folders/4s/kj0rwqjj67sfjnbv5387s2fw0000gp/T/go-build2327985354/b499/_x004.o -x c++ bind.cc
95098   0.0  0.0 407962448     48 s023  S+   12:01PM   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox clang
95014   0.0  0.0 408212720  11712 s032  S+   12:00PM   0:00.01 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -I /go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/go-zetasql -fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/4s/kj0rwqjj67sfjnbv5387s2fw0000gp/T/go-build2327985354/b499=/tmp/go-build -gno-record-gcc-switches -fno-common -I /var/folders/4s/kj0rwqjj67sfjnbv5387s2fw0000gp/T/go-build2327985354/b499/ -O2 -g -std=c++1z -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/protobuf -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/gtest -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/icu -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/re2 -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/json -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/googleapis -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/flex/src -Wno-char-subscripts -Wno-sign-compare -Wno-switch -Wno-unused-function -Wno-deprecated-declarations -Wno-inconsistent-missing-override -Wno-unknown-attributes -Wno-macro-redefined -Wno-shift-count-overflow -Wno-enum-compare-switch -Wno-return-type -Wno-subobject-linkage -Wno-unknown-warning-option -DHAVE_PTHREAD -DU_COMMON_IMPLEMENTATION -std=c++1z -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/protobuf -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/gtest -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/icu -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/re2 -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/json -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/googleapis -I/go/pkg/mod/github.com/goccy/go-zetasql@v0.5.1/internal/ccall/flex/src -Wno-char-subscripts -Wno-sign-compare -Wno-switch -Wno-unused-function -Wno-deprecated-declarations -Wno-inconsistent-missing-override -Wno-unknown-attributes -Wno-macro-redefined -Wno-shift-count-overflow -Wno-enum-compare-switch -Wno-return-type -Wno-subobject-linkage -Wno-unknown-warning-option -DHAVE_PTHREAD -DU_COMMON_IMPLEMENTATION -o /var/folders/4s/kj0rwqjj67sfjnbv5387s2fw0000gp/T/go-build2327985354/b499/_x004.o -c bind.cc
```

Reply:
Able to observe the same issue on my system as well.